### PR TITLE
#38 Added support for bigQuery in load_file() operator

### PR DESCRIPTION
**Pull request by utkarsharma2**
1. Added support for bigQuery in load_file() operator.
2. Replaced to_sql() with to_gbq() due to a bug within pandas to_sql() function.
3. Fixed BigQueryHook issue with a work around Temp hook.

### CODE REVIEW

File: src/astro/sql/operators/temp_hooks.py [hunk 1]
```diff
@@ -56,3 +61,46 @@ def get_uri(self) -> str:
         if self.schema:
             uri += self.schema
         return uri
+
+
+class TempBigQueryHook(BigQueryHook):
```
Comment (dimberman) [inline src/astro/sql/operators/temp_hooks.py:66]: @utkarsharma2 can you please document the bug you are working around so we can remove this once the upstream is fixed?

File: src/astro/utils/load_dataframe.py [hunk 2]
```diff
@@ -44,6 +49,12 @@ def move_dataframe_to_sql(
             schema=schema,
             warehouse=warehouse,
         ),
+        "google_cloud_platform": BigQueryHook(
+            gcp_conn_id=conn_id, use_legacy_sql=False
+        ),
+        "bigquery": TempBigQueryHook(
+            bigquery_conn_id=conn_id, use_legacy_sql=False, gcp_conn_id=conn_id
+        ),
```
Comment (dimberman) [inline src/astro/utils/load_dataframe.py:57]: What is the difference here?

**Review by dimberman — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 3]
```diff
@@ -18,13 +18,13 @@ dependencies = [
...
     "apache-airflow-providers-postgres",
     "apache-airflow-providers-snowflake",
+    "apache-airflow-providers-google",
     "pandas >=1.3.4",
     "s3fs",
     "snowflake-sqlalchemy ==1.2.0",
     "snowflake-connector-python[pandas]",
     "smart-open[all]>=5.2.1",
     "SQLAlchemy == 1.4.28",
-    "snowflake-sqlalchemy"
```
Comment (dimberman) [inline pyproject.toml:27]: Why did you remove this:?

File: pyproject.toml [hunk 4]
```diff
@@ -18,13 +18,13 @@ dependencies = [
     "apache-airflow >=2.0",
     "apache-airflow-providers-postgres",
     "apache-airflow-providers-snowflake",
+    "apache-airflow-providers-google",
     "pandas >=1.3.4",
     "s3fs",
     "snowflake-sqlalchemy ==1.2.0",
     "snowflake-connector-python[pandas]",
     "smart-open[all]>=5.2.1",
     "SQLAlchemy == 1.4.28",
```
Comment (dimberman) [inline pyproject.toml:27]: @ashb do we need to downgrade this?

**Review by dimberman — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (dimberman) [reply pyproject.toml:27]: Oh I see the double import nvm

File: src/astro/utils/load_dataframe.py (hunk 2 above)
Comment (utkarsharma2) [reply src/astro/utils/load_dataframe.py:57]: Ohh we don't need 'google_cloud_platform' removing it.

File: src/astro/sql/operators/temp_hooks.py (hunk 1 above)
Comment (utkarsharma2) [reply src/astro/sql/operators/temp_hooks.py:66]: Sure, will add a comment.

File: pyproject.toml (hunk 4 above)
Comment (utkarsharma2) [reply pyproject.toml:27]: I think we need to downgrade it to Airflow's sqlAlchemy version - 1.3, because with latest pip we cannot have conflicting dependencies.

File: pyproject.toml [hunk 5]
```diff
@@ -18,11 +18,16 @@ dependencies = [
...
+    "apache-airflow-providers-google",
+    "python-frontmatter",
     "pandas >=1.3.4",
     "s3fs",
     "snowflake-sqlalchemy ==1.2.0",
     "snowflake-connector-python[pandas]",
     "smart-open[all]>=5.2.1",
+    "SQLAlchemy == 1.4.28",
+    "snowflake-sqlalchemy",
+    "sqlalchemy-bigquery"
```
Comment (tatiana) [inline pyproject.toml:30]: @utkarsharma2 @dimberman I'd love to get the view of both of you on our approach towards Astro's dependencies

I just created isolated virtualenvs for both `main` and `SQLAlchemy-demo` branches by running:
```
pip install .[dependencies]
```

Some of the differences were that in `SQLAlchemy-demo` we downgraded a few packages:
- `apache-airflow==2.2.3` -> `apache-airflow==2.0.1`
- `apache-airflow-providers-postgres==2.4.0` -> `apache-airflow-providers-postgres==1.0.2 `
- `apache-airflow-providers-postgres==2.4.0` -> `apache-airflow-providers-snowflake==1.3.0`

Was this downgrade be intentional? Could this affect Astro's behaviour/adoption in any sense?

 Also, in `main`, `pip freeze` lists 145 dependencies, and the branch   `SQLAlchemy-demo` has 192. Are we happy for Astro to have this amount of dependencies, or do we value fewer dependencies? There are successful open source projects which followed both paths of lots of dependencies / minimal dependencies (e.g. Plone and Django) - it would be great to understand where we stand.

File: src/astro/sql/operators/sql_decorator.py [hunk 6]
```diff
@@ -94,23 +95,24 @@ def execute(self, context: Dict):
         self.run_id = context.get("run_id")
         self.convert_op_arg_dataframes()
         self.convert_op_kwarg_dataframes()
-        if not self.sql:
+        if self.sql is None or self.sql == "":
```
Comment (tatiana) [inline src/astro/sql/operators/sql_decorator.py:98]: @utkarsharma2 it may be worth adding a comment why we are not doing `not self.sql` - people less familiar with SQLAlchemy might be tempted to write as it was before

File: src/astro/sql/operators/temp_hooks.py [hunk 7]
```diff
@@ -28,7 +33,7 @@ def get_uri(self) -> str:
         """Override DbApiHook get_uri method for get_sqlalchemy_engine()"""
         conn_config = self._get_conn_params()
         uri = (
-            "snowflake://{user}:{password}@{account}.{region}/{database}/{schema}"
```
Comment (tatiana) [inline src/astro/sql/operators/temp_hooks.py:31]: Does dropping `region` have any implications?

**Review by tatiana — COMMENTED** [review]
Great progress on supporting BQ, @utkarsharma2 !
I left a few questions/comments inline.
What do you think if we had this PR based on `main` as opposed to `SQLAlchemy` branch?

File: tests/operators/test_agnostic_load_file.py [hunk 8]
```diff
@@ -466,47 +497,84 @@ def test_aql_gcs_file_to_postgres(self):
...
+    },
+)
+def test_aws_decode():
+    from astro.utils.cloud_storage_creds import parse_s3_env_var
+
+    k, v = parse_s3_env_var()
+    assert v == "@#$%@$#ASDH@Ksd23%SD546"
+
+
+@pytest.mark.parametrize("sql_server", ["snowflake", "postgres"], indirect=True)
```
Comment (tatiana) [inline tests/operators/test_agnostic_load_file.py:551]: @utkarsharma2 what do you think of adding BQ to this test?

Comment (dimberman) [reply tests/operators/test_agnostic_load_file.py:551]: I think that sounds like a good idea.

File: README.md [hunk 9]
```diff
@@ -42,19 +42,25 @@
 </h3>
 <br/>
 
+# Overview
```
Comment (dimberman) [inline README.md:45]: @utkarsharma2 can you fix this so that there aren't conflicts with the REDME on main?

**Review by dimberman — CHANGES_REQUESTED** [review]
